src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -141927,5 +141927,652 @@ static T GetValue2<T>(Collection c, object key)
+            var item = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().First();
+            Assert.Equal("i.Get()", item.ToString());
+            Assert.Equal("ref System.Object I<System.Object>.Get()", model.GetSymbolInfo(item).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithoutConstraint_ReverseOrder()
+        {
+            var src = @"
+#nullable disable
+interface I<T> { T Item { get; set; } }
+
+#nullable enable
+interface I2<T> : I<T> { }
+
+#nullable disable
+interface I3 : I2<object>, I<object> { }
+
+#nullable enable
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+        i.Item = null;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+            var i3 = comp.GetTypeByMetadataName("I3");
+            AssertEx.Equal(new string[] { "I2<object>", "I<object>" },
+                i3.Interfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+            AssertEx.Equal(new string[] { "I2<object>", "I<object>" },
+                i3.AllInterfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var item = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().First();
+            Assert.Equal("i.Item", item.ToString());
+            Assert.Equal("System.Object I<System.Object>.Item { get; set; }", model.GetSymbolInfo(item).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithConstraint()
+        {
+            var src = @"
+#nullable disable
+interface I<T> where T : class { T Item { get; set; } }
+
+#nullable enable
+interface I2<T> : I<T> where T : class { }
+
+#nullable disable
+interface I3 : I<object>, I2<object> { }
+
+#nullable enable
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+        i.Item = null;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+            var i3 = comp.GetTypeByMetadataName("I3");
+            AssertEx.Equal(new string[] { "I<object>", "I2<object>" },
+                i3.Interfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+            AssertEx.Equal(new string[] { "I<object>", "I2<object>", "I<object!>" },
+                i3.AllInterfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var item = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().First();
+            Assert.True(model.LookupNames(item.SpanStart, i3.GetPublicSymbol()).Contains("Item"));
```

Comment:
>.Contains("Item") [](start = 79, length = 17)

I think we actually want to assert in a way that would allow us to verify from which flavor of the interface this symbol actually came from. We want to see is there an ambiguity here and how it is resolved.